app/assets/javascripts/components/common/ArticleViewer/containers/ArticleViewer.jsx

```diff
@@ -63,9 +63,13 @@ export class ArticleViewer extends React.Component {
   // first in that case. In that case, componentDidUpdate fetches the
   // user ids as soon as usernames are avaialable.
   componentDidUpdate(prevProps, prevState) {
-    if (!prevProps.users && this.props.users) {
-      if (!prevState.userIdsFetched) {
-        this.fetchUserIds(this.props.users);
+    if (this.props.assignedUsers && this.props.users) {
```

Comment:
It would be helpful to have comments explaining these two different branches where `fetchUserIds` can happen and why it needs to be branched like this. From a user perspective, what are the different situations where each of these branches occurs?